lsif/src/shared/tasks.ts

```diff
@@ -0,0 +1,50 @@
+import AsyncPolling from 'async-polling'
+import { Connection } from 'typeorm'
+import { logAndTraceCall, TracingContext } from './tracing'
+import { Logger } from 'winston'
+import { tryWithLock } from './store/locks'
+
+interface Task {
+    intervalMs: number
+    handler: () => Promise<void>
+}
+
+/** A collection of tasks that are invoked periodically. */
+export class TaskRunner {
```

Comment:
The name `TaskRunner` is quite generic, but the implementation assumes it's a particular _kind_ of task (viz., one that wants a lock on a specific Postgres connection). A couple things I'd like to recommend here:

1. The runner should pass the connection to the task, rather than assuming the task function will close over it from elsewhere. That will help reduce the risk that tasks will capture a different database handle than the one where we're doing the locking.

2. The class comment should mention something about the constraint. Maybe something like:

```suggestion
/** A collection of tasks that are invoked periodically, each holding an advisory
    exclusive lock on a PostgreSQL database connection.
  */
export class DatabaseTaskRunner {
```

or words to that effect.

Reply:
> The runner should pass the connection to the task

Not all tasks touch the db, some only do some file stuff that shouldn't be done concurrently by two processes. It doesn't hurt to pass it, though.

Renaming to `ExclusivePeriodicTaskRunner`.